frameworks/autosklearn/exec.py

```diff
@@ -50,45 +51,64 @@ def run(dataset, config):
+            min(
+                config.max_mem_size_mb / n_jobs,
+                math.ceil(total_memory_mb / n_jobs)
+            ),
+            3072  # 3072 is autosklearn default and we use it as a lower bound
+        )
+    if askl_version >= version.parse("0.11"):
+        log.info(
+            "Using %sMB memory per job and on a total of %s jobs.",
+            ml_memory_limit, n_jobs
+        )
+        constr_params["memory_limit"] = ml_memory_limit
+    else:
+        ensemble_memory_limit = config.framework_params.get('_ensemble_memory_limit', 'auto')
+        # when memory is large enough, we should have:
+        # (cores - 1) * ml_memory_limit_mb + ensemble_memory_limit_mb = config.max_mem_size_mb
+        total_memory_mb = utils.system_memory_mb().total
+        if ml_memory_limit == 'auto':
+            ml_memory_limit = max(min(config.max_mem_size_mb,
+                                      math.ceil(total_memory_mb / n_jobs)),
+                                  3072)  # 3072 is autosklearn defaults
+        if ensemble_memory_limit == 'auto':
+            ensemble_memory_limit = max(math.ceil(ml_memory_limit - (total_memory_mb - config.max_mem_size_mb)),
+                                        math.ceil(ml_memory_limit / 3),  # default proportions
+                                        1024)  # 1024 is autosklearn defaults
+        log.info("Using %sMB memory per ML job and %sMB for ensemble job on a total of %s jobs.", ml_memory_limit, ensemble_memory_limit, n_jobs)
+        constr_params["ml_memory_limit"] = ml_memory_limit
+        constr_params["ensemble_memory_limit"] = ensemble_memory_limit
 
     log.warning("Using meta-learned initialization, which might be bad (leakage).")
     # TODO: do we need to set per_run_time_limit too?
     estimator = AutoSklearnClassifier if is_classification else AutoSklearnRegressor
 
-    if version.parse(autosklearn.__version__) >= version.parse("0.8"):
-        constr_extra_params = dict(metric=perf_metric)
-        fit_extra_params = {}
+    if askl_version >= version.parse("0.8"):
+        constr_params['metric'] = perf_metric
     else:
-        constr_extra_params = {}
-        fit_extra_params = dict(metric=perf_metric)
-
-    auto_sklearn = estimator(time_left_for_this_task=config.max_runtime_seconds,
-                             n_jobs=n_jobs,
-                             ml_memory_limit=ml_memory_limit,
-                             ensemble_memory_limit=ensemble_memory_limit,
-                             seed=config.seed,
-                             **constr_extra_params,
-                             **training_params)
+        fit_extra_params['metric'] = perf_metric
+
+    constr_params["time_left_for_this_task"] = config.max_runtime_seconds
+    constr_params["n_jobs"] = n_jobs
+    constr_params["seed"] = config.seed
+
+    log.info("Auto-sklearn constructor arguments: %s", constr_params)
+    log.info("Auto-sklearn fit() arguments: %s", training_params)
```

Comment:
`fit() arguments` != `training_params`
the latter are additional optional params that can be passed in the framework definition.

you probably meant `fit_extra_params`

Reply:
Yes, indeed. I will update this locally, test locally, and then update this PR.